### PR TITLE
Using structs plugin

### PR DESCRIPTION
**Pull request by kohsuke**
`DescribableHelper` functionality has moved to structs plugin, in preparation for the death of `$class`

### CODE REVIEW

**Comment by kohsuke** [comment]
I talked to Jesse offline about this, but otherwise @reviewbybees

File: step-api/pom.xml [hunk 1]
```diff
@@ -41,6 +41,11 @@
         </dependency>
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
+            <artifactId>structs</artifactId>
+            <version>1.0-20160312.085007-3</version>
```
Comment (abayer) [inline step-api/pom.xml]: I assume there'll be an actual release before Pipeline's next release, yes? =)

Comment (kohsuke) [inline step-api/pom.xml]: Before this merges, yes. This is considered a part of the exit criteria for the structs plugin.

Comment (abayer) [inline step-api/pom.xml]: Then I am :bee:

File: step-api/src/main/java/org/jenkinsci/plugins/workflow/structs/DescribableHelper.java [hunk 2]
```diff
@@ -73,6 +73,9 @@
 /**
  * Utility for converting between {@link Describable}s (and some other objects) and map-like representations.
  * Ultimately should live in Jenkins core (or Stapler).
+ *
+ * @deprecated
+ *      Moved to the structs plugin
```
Comment (amuniz) [inline step-api/src/main/java/org/jenkinsci/plugins/workflow/structs/DescribableHelper.java]: It seems it is not really moved but moved-renamed (it is now `DescribableModel `). Maybe a link to the new class in the javadoc makes sense (and developers will be thankful!)

**Comment by jglick** [comment]
On my list to review…

File: step-api/src/main/java/org/jenkinsci/plugins/workflow/structs/DescribableHelper.java (hunk 2 above)
Comment (jglick) [inline step-api/src/main/java/org/jenkinsci/plugins/workflow/structs/DescribableHelper.java]: Yes ought to be using `{@link DescribableModel}`.

File: step-api/src/main/java/org/jenkinsci/plugins/workflow/steps/StepDescriptor.java [hunk 3]
```diff
@@ -110,7 +111,7 @@ public Step newInstance(Map<String,Object> arguments) throws Exception {
      * @throws UnsupportedOperationException if this descriptor lacks the ability to do such a calculation
      */
     public Map<String,Object> defineArguments(Step step) throws UnsupportedOperationException {
-        return DescribableHelper.uninstantiate(step);
+        return DescribableModel.uninstantiate_(step);
```
Comment (jglick) [inline step-api/src/main/java/org/jenkinsci/plugins/workflow/steps/StepDescriptor.java]: :ant: an oddly-named method. I would just delete it. Here it is not cumbersome to write

```java
return new DescribableModel(step.getClass()).uninstantiate(step);
```

and for the usage in `InputStepTest` it is easy enough to hardcode `InputStep.class`.

Comment (jglick) [inline step-api/src/main/java/org/jenkinsci/plugins/workflow/steps/StepDescriptor.java]: (Seems your initial code drop in `symbol-plugin` did not go through the PR process so there is no easy way to comment on that code directly. Generally I would suggest starting with an empty repository and then filing PR 1 to add initial proposed content.)

**Comment by jglick** [comment]
:bee:

**Comment by kohsuke** [comment]
Feedbacks to incorporate:

- [ ] remove `uninstantiate_`
- [ ] javadoc fix

**Comment by kohsuke** [comment]
On 2nd thought I'm keeping `uninstantiate_` method because the instance method version is fundamentally redundant by taking two arguments. The underscore in the end and not in the beginning is to assist auto-completion discoverability.

It's up to calling code to decide what method to call, so if it bothers you, feel free to use the non-static version.

**Comment by jglick** [comment]
@kohsuke you seem to think you released 1.0, yet you did not actually do so. There is no such release in the repository, and https://github.com/jenkinsci/structs-plugin/commits/master shows no release commit, though there is a https://github.com/jenkinsci/structs-plugin/tree/structs-parent-1.0 tag.

**Comment by jglick** [comment]
:bee:

**Pull request merged**